src/scheduler.cc

```diff
@@ -702,6 +723,40 @@ void SchedulerService::get_equivalent_objrefs(ObjRef objref, std::vector<ObjRef>
   upstream_objrefs(downstream_objref, equivalent_objrefs);
 }
 
+// This method (along with release_all_locks) defines the order in which locks should be acquired.
+void SchedulerService::acquire_all_locks() {
+  pull_queue_lock_.lock();
+  computation_graph_lock_.lock();
+  fntable_lock_.lock();
+  avail_workers_lock_.lock();
+  task_queue_lock_.lock();
+  alias_notification_queue_lock_.lock();
+  workers_lock_.lock();
+  reference_counts_lock_.lock();
+  contained_objrefs_lock_.lock();
+  objects_lock_.lock();
+  objstores_lock_.lock();
+  target_objrefs_lock_.lock();
+  reverse_target_objrefs_lock_.lock();
+}
+
+// This method (along with acquire_all_locks) defines the order in which locks should be acquired.
+void SchedulerService::release_all_locks() {
+  pull_queue_lock_.unlock();
+  computation_graph_lock_.unlock();
+  fntable_lock_.unlock();
+  avail_workers_lock_.unlock();
+  task_queue_lock_.unlock();
+  alias_notification_queue_lock_.unlock();
+  workers_lock_.unlock();
+  reference_counts_lock_.unlock();
+  contained_objrefs_lock_.unlock();
+  objects_lock_.unlock();
+  objstores_lock_.unlock();
+  target_objrefs_lock_.unlock();
+  reverse_target_objrefs_lock_.unlock();
+}
+
```

Comment:
DRY (don't repeat yourself).
Factor these out into a single array of member pointers, and then iterate over them to lock/unlock.